cmake_toolchain/vita.cmake

```diff
@@ -40,7 +60,7 @@ include(CMakeParseArguments)
 ## @param[opt] CONFIG file
 ##   Path to a YAML config file defining exports and other optional information
 ##
-macro(vita_create_self target source)
+macro(vita_create_self_target target self source)
```

Comment:
could you change the macro name `vita_create_self_target` to `_vita_create_self_target` to make the filling of the `private`?
if someone use this macro to their cmake rule, we should keep this compatibility in future like these whole bug chains

Reply:
so you want to provide no options for make builds to avoid circular dependencies?

Reply:
well I just want to fix the problem. and I don't want to give multiple way to building packages to users.
if they want it, they should use `manual way` with `add_custom_command` and `add_custom_target` or they have to follow the `vitasdk`'s macros (eg; `vita_create_self`, `vita_create_stubs` and `vita_create_vpk`)
so I don't want to export these new macro.
(idk these can have `_` prefix, if we are able to use this like `_vita_create_self_target`, nobody think that this macro is public.)